openhands/llm/retry_mixin.py

```diff
@@ -12,28 +14,41 @@
 class RetryMixin:
     """Mixin class for retry logic."""
 
-    def retry_decorator(self, **kwargs):
+    def retry_decorator(self, **kwargs: Any):
         """
         Create a LLM retry decorator with customizable parameters. This is used for 429 errors, and a few other exceptions in LLM classes.
 
         Args:
             **kwargs: Keyword arguments to override default retry behavior.
-                      Keys: num_retries, retry_exceptions, retry_min_wait, retry_max_wait, retry_multiplier
+                      Keys: num_retries, retry_exceptions, exclude_exceptions, retry_min_wait, retry_max_wait, retry_multiplier
 
         Returns:
             A retry decorator with the parameters customizable in configuration.
         """
         num_retries = kwargs.get('num_retries')
-        retry_exceptions = kwargs.get('retry_exceptions')
+        retry_exceptions = kwargs.get('retry_exceptions', ())
+        exclude_exceptions = kwargs.get('exclude_exceptions', ())
         retry_min_wait = kwargs.get('retry_min_wait')
         retry_max_wait = kwargs.get('retry_max_wait')
         retry_multiplier = kwargs.get('retry_multiplier')
 
+        def _should_retry(exception: Exception) -> bool:
+            """Check if an exception should be retried based on its type and exclusion list."""
+            # First check if it's an excluded exception
+            for exc_type in exclude_exceptions:
+                if isinstance(exception, exc_type):
+                    return False
+            # Then check if it's a retryable exception
+            for exc_type in retry_exceptions:
+                if isinstance(exception, exc_type):
+                    return True
+            return False
+
         return retry(
             before_sleep=self.log_retry_attempt,
             stop=stop_after_attempt(num_retries) | stop_if_should_exit(),
             reraise=True,
-            retry=(retry_if_exception_type(retry_exceptions)),
+            retry=retry_if_exception(_should_retry),
```

Comment:
@openhands-agent You're doing unnecessary stuff in retry_mixin.py. Think about the tenacity library: you know tenacity, and you know it has already defined retry_if_exception_type and retry_if_not_exception_type.

So use the existing features of tenacity, instead of a your custom "_should_retry" function.